src/hbb/utils.py

```diff
@@ -0,0 +1,278 @@
+    data_dir: Path,
+    process: str,
+    samples: list[str],
+    year: str,
+    filters: list = None,
+    columns: list = None,
+    load_weight_noxsec: bool = True,
+) -> dict[str, pd.DataFrame]:
+    """
+    Loads events with an optional filter.
+    Divides MC samples by the total pre-skimming, to take the acceptance into account.
+
+    Args:
+        data_dir (str): path to data directory.
+        samples (List[str]): list of sample names to load.
+        year (str): year.
+        filters (List): Optional filters when loading data.
+        columns (List): Optional columns to load.
+
+    Returns:
+        Dict[str, pd.DataFrame]: Dictionary of events dataframe for each sample.
+    """
+    data_dir = Path(data_dir) / year
+    full_samples_list = listdir(data_dir)  # get all directories in data_dir
+    events_dict = {}
+
+    for sample_name in samples:
+        # Initialize the events list for the sample
+        print("sample_name", sample_name)
+        events_list = []
+        # Check if sample directory exists in full_samples_list
+        for sample in full_samples_list:
+            if not check_selector(sample, sample_name):
+                continue
+
+            sample_path = data_dir / sample
+            parquet_path, pickles_path = sample_path / "parquet", sample_path / "pickles"
+
+            # No parquet directory?
+            if not parquet_path.exists():
+                warnings.warn(f"No parquet directory for {sample}!", stacklevel=1)
+                continue
+
+
+            print(f"Loading {sample}")
+            events = pd.read_parquet(parquet_path, filters=filters, columns=columns)
+
+            # No events?
+            if not len(events):
+                warnings.warn(f"No events for {sample}!", stacklevel=1)
+                continue
+
+            # Normalize by total events
+            if process != "data":
+                n_events = get_nevents(pickles_path, year, sample)
+                events["weight_nonorm"] = events["weight"]
+                events["finalWeight"] = events["weight"] / n_events
+                #print(events["finalWeight"])
+            else:
+                events["finalWeight"] = events["weight"]
+
+            events_list.append(events)
+            print(f"Loaded {sample: <50}: {len(events)} entries")
+
+        # Combine all DataFrames for the sample
+        if events_list:
+            events_dict[sample_name] = pd.concat(events_list)
+        else:
+            warnings.warn(f"No valid events loaded for sample {sample_name}.", stacklevel=1)
+
+    return events_dict
+
+
+
+
+def format_columns(columns: list):
+    """
+    Reformat input of (`column name`, `num columns`) into (`column name`, `idx`) format for
+    reading multiindex columns
+    """
+    ret_columns = []
+    for key, num_columns in columns:
+        for i in range(num_columns):
+            ret_columns.append(f"('{key}', '{i}')")
+    return ret_columns
+
+
+def diffload_samples(
```

Comment:
what is this function used for? how does it differ from load_samples? Keep only the function you intend to use